ambari-server/src/test/java/org/apache/ambari/server/serveraction/kerberos/KerberosTopologyUpdateTriggerServerActionTest.java

```diff
@@ -0,0 +1,62 @@
+package org.apache.ambari.server.serveraction.kerberos;
+
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.verify;
+import static org.mockito.Matchers.anyBoolean;
+
+import org.apache.ambari.server.agent.stomp.TopologyHolder;
+import org.apache.ambari.server.audit.AuditLogger;
+import org.apache.ambari.server.events.TopologyUpdateEvent;
+import org.apache.ambari.server.testutils.PartialNiceMockBinder;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockSupport;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+
+public class KerberosTopologyUpdateTriggerServerActionTest extends EasyMockSupport {
+
+  private static TopologyHolder topologyHolder = EasyMock.createNiceMock(TopologyHolder.class);
+
+  private KerberosTopologyUpdateTriggerServerAction action;
+
+  @Before
+  public void init() {
+    final Injector injector = createInjector();
+    action = injector.getInstance(KerberosTopologyUpdateTriggerServerAction.class);
+  }
+
+  @Test
+  public void shouldUpdateTopology() throws Exception {
+    final TopologyUpdateEvent event = createNiceMock(TopologyUpdateEvent.class);
+    expect(topologyHolder.getCurrentData()).andReturn(event).once();
+    expect(topologyHolder.updateData(event)).andReturn(anyBoolean()).once();
+    replay(topologyHolder);
+    action.execute(null);
+    verify(topologyHolder);
+  }
+
+  static class TestTopologyHolderProvider implements Provider<TopologyHolder> {
+    @Override
+    public TopologyHolder get() {
+      return topologyHolder;
+    }
+  }
+
+  private Injector createInjector() {
```

Comment:
Would it be possible to test this without guice, just by passing the dependencies via constructor?

Reply:
We need Guice anyway...either the field would be injected or the constructor...how else would you get `TopologyHolder`? Please note this a Singleton scoped Guice bean.

Reply:
Why do we need guice to do this?

```java
action = new KerberosTopologyUpdateTriggerServerAction(topologyHolder);
```

Reply:
I meant we need Guice anyway in order to retrieve `TopologyHolder` when running on the server. On the other hand it makes sense to pass it as a constructor argument (which allows unit testing with simple POJOs/Mocks) and let Guice inject the constructor within the server. Please review it again.